unified-runtime/source/adapters/opencl/event.cpp

```diff
@@ -149,11 +149,29 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventRetain(ur_event_handle_t hEvent) {
 
 UR_APIEXPORT ur_result_t UR_APICALL
 urEventWait(uint32_t numEvents, const ur_event_handle_t *phEventWaitList) {
-  std::vector<cl_event> CLEvents(numEvents);
+  if (numEvents == 0 || !phEventWaitList) {
```

Comment:
either of these conditions are invalid according to the spec, is sycl passing `0`/`nullptr`?

Reply:
No, you're right, this is not needed. Done.